src/cpp/src/scheduler.hpp

```diff
@@ -34,11 +33,14 @@ class Scheduler {
     };
 
     explicit Scheduler(const SchedulerConfig & config = {}) :
-        m_config(config), m_block_manager(m_config.num_kv_blocks) { }
+        m_config(config), m_block_manager(m_config.num_kv_blocks, m_config.enable_prefix_caching, m_config.block_size) { }
 
     Output schedule(std::vector<SequenceGroup::Ptr>& sequence_groups) {
         Output scheduler_output;
 
+        if (m_config.enable_prefix_caching)
+            _restore_cached_blocks(sequence_groups);
```

Comment:
do we need to perform it on each iteration? As far as I could remember, vLLM does this prefix caching during creation of sequence groups.